DeepSpeech.py

```diff
@@ -481,51 +445,51 @@ def train():
                 sys.exit(1)
 
         # Retrieving global_step from restored model and setting training parameters accordingly
-        model_feeder.set_data_set(no_dropout_feed_dict, train_set)
-        step = session.run(global_step, feed_dict=no_dropout_feed_dict)
+        step = session.run(global_step)
         num_gpus = len(Config.available_devices)
-        steps_per_epoch = max(1, train_set.total_batches // num_gpus)
-        steps_trained = step % steps_per_epoch
+        steps_per_epoch = max(1, train_batches // num_gpus)
         current_epoch = step // steps_per_epoch
         target_epoch = current_epoch + abs(FLAGS.epoch) if FLAGS.epoch < 0 else FLAGS.epoch
-        train_index.index = steps_trained * num_gpus
 
         log_debug('step: %d' % step)
         log_debug('epoch: %d' % current_epoch)
         log_debug('target epoch: %d' % target_epoch)
         log_debug('steps per epoch: %d' % steps_per_epoch)
         log_debug('batches per step (GPUs): %d' % num_gpus)
-        log_debug('number of batches in train set: %d' % train_set.total_batches)
-        log_debug('number of batches already trained in epoch: %d' % train_index.index)
+        log_debug('number of batches in train set: %d' % train_batches)
 
-        def run_set(set_name):
-            data_set = getattr(model_feeder, set_name)
+        def run_set(set_name, init_op, num_batches):
```

Comment:
nit: Regarding `init_op`: Why not going the dictionary approach - like for the summary writers?

Reply:
To avoid more coupling of different parts of the code via strings.

Reply:
I stated this from a symmetry perspective - either doing it the one or the other way.